src/config/gobjects.rs

```diff
@@ -236,8 +237,11 @@ fn parse_object(
         .map(|s| s.to_owned());
     let generate_trait = toml_object
         .lookup("trait")
+        .and_then(|v| v.as_bool());
+    let final_type = toml_object
+        .lookup("final_type")
         .and_then(|v| v.as_bool())
-        .unwrap_or(true);
+        .or(generate_trait.map(|t| !t));
```

Comment:
Clippy suggest `.or_else` and some other nits

Reply:
What other nits?

I thought about using `or_else()` here but it seemed more effort than inverting a boolean would justify. Your decision if I should change it or not :)

Reply:
You right, but then need adding allow for `clippy::or_fun_call`.
Other warnings (strange that only first match/Class listed IMHO it same for both) :
```
warning: redundant pattern matching, consider using `is_some()`
   --> src\config\gobjects.rs:288:12
    |
288 |       if let Some(_) = generate_trait {
    |  _____-      ^^^^^^^
289 | |         warn!("`trait` configuration is deprecated and replaced by `final_type` for object {}", name);
290 | |     }
    | |_____- help: try this: `if generate_trait.is_some()`
    |
    = note: #[warn(clippy::redundant_pattern_matching)] on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pattern_matching

warning: you seem to be trying to use match for destructuring a single pattern. Consider using `if let`
   --> src\library_postprocessing.rs:392:17
    |
392 | /                 match *type_ {
393 | |                     Type::Class(ref klass) => {
394 | |                         let tid = TypeId {
395 | |                             ns_id: ns_id as u16,
...   |
413 | |                     _ => (),
414 | |                 }
    | |_________________^
    |
    = note: #[warn(clippy::single_match)] on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_match
help: try this
    |
392 |                 if let Type::Class(ref klass) = *type_ {
393 |     let tid = TypeId {
394 |         ns_id: ns_id as u16,
395 |         id: id as u32,
396 |     };
397 | 
```

Reply:
Ok, changing :)

Reply:
Pushed